processing/src/main/java/io/druid/query/topn/PooledTopNAlgorithm.java

```diff
@@ -311,12 +311,14 @@ protected void closeAggregators(BufferAggregator[] bufferAggregators)
   @Override
   public void cleanup(PooledTopNParams params)
   {
-    ResourceHolder<ByteBuffer> resultsBufHolder = params.getResultsBufHolder();
+    if (params != null) {
```

Comment:
The original interface could be annotated with `@Nullable`, that would make any extending implementation able to handle this failure case.